acceptance/tests/ticket_4622_filebucket_diff_test.rb

```diff
@@ -1,86 +1,72 @@
 test_name "ticket 4622 filebucket diff test."
 confine :except, :platform => 'windows'
+skip_test 'skip test, no non-Windows agents specified' if agents.empty?
 
-def create_testfile(filename, contents)
-  create_remote_file(agent, filename, "#{contents}")
-end
-
-def get_checksum_from_backup(filename, bucket_locale)
-  on agent, puppet("filebucket backup #{filename} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
+def get_checksum_from_backup_on(host, filename, bucket_locale)
+  on host, puppet("filebucket backup #{filename} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
   output = result.stdout.strip.split(": ")
   checksum = output.last
   return checksum
 end
 
-def validate_diff(item_one, item_two, bucket_locale)
-  on agent, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
+def validate_diff_on(host, item_one, item_two, bucket_locale)
+  on host, puppet("filebucket diff #{item_one} #{item_two} #{bucket_locale}"), :acceptable_exit_codes => [ 0, 2 ]
   assert_match("-foo", result.stdout.strip)
   assert_match('+bar', result.stdout.strip)
   assert_match('+baz', result.stdout.strip)
```

Comment:
Could work on another PR to clean it up. I was thinking of trying to define a structure that contains values to add and remove, then helpers to modify the file or assert diffs based on that structure.